src/canvas/SelectableCanvas.ts

```diff
@@ -754,7 +755,49 @@ export class SelectableCanvas<EventSpec extends CanvasEvents = CanvasEvents>
   }
 
   /**
-   * Checks point is inside the object.
+   * Checks if the point is inside the object selection area including padding
+   * @param {FabricObject} obj Object to test against
+   * @param {Object} [pointer] point in scene coordinates
+   * @return {Boolean} true if point is contained within an area of given object
+   * @private
+   */
+  private _pointIsInObjectSelectionArea(obj: FabricObject, point: Point) {
+    // getCoords will already take care of group de-nesting
+    let coords = obj.getCoords();
+    const viewportZoom = this.getZoom();
+    const padding = obj.padding / viewportZoom;
+    if (padding) {
+      const [tl, tr, br, bl] = coords;
+      // what is the angle of the object?
+      // we could use getTotalAngle, but is way easier to look at it
+      // from how coords are oriented, since if something went wrong
+      // at least we are consistent.
+      const angleRadians = Math.atan2(tr.y - tl.y, tr.x - tl.x),
+        cosP = cos(angleRadians) * padding,
+        sinP = sin(angleRadians) * padding,
+        cosPSinP = cosP + sinP,
+        cosPMinusSinP = cosP - sinP;
+
+      coords = [
+        new Point(tl.x - cosPMinusSinP, tl.y - cosPSinP),
+        new Point(tr.x + cosPSinP, tr.y - cosPMinusSinP),
+        new Point(br.x + cosPMinusSinP, br.y + cosPSinP),
+        new Point(bl.x - cosPSinP, bl.y + cosPMinusSinP),
+      ];
+      // in case of padding we calculate the new coords on the fly.
+      // otherwise we have to maintain 2 sets of coordinates for everything.
+      // we can reiterate on storing those on something similar to lineCoords
+      // if this is slow, for now the semplification is large and doesn't impact
+      // rendering.
+      // the idea behind this is that outside target check we don't need ot know
+      // where those coords are
+    }
+    return Intersection.isPointInPolygon(point, coords);
+  }
+
```

Comment:
Do you mind if I simplify the logic here using a padding vector?
Does it have to be private?
Can we instead change `containsPoint` to accept a 3rd param `padding` defaulting to 0?
I think that is much better both for fabric and for the dev

Reply:
my issue here is that Intersection is not a selection thing.
This is a method entirely done to check the selection and padding is selection only.
Why would you use containsPoint with padding? use checkTarget right away no?
if is padding is a selection thing.

Why would you use containsPoint instead of checkTarget?


Regarding private, i did that because i thought, let's see how it shapes out, become public is easier, or maybe then we need to change it radically or change its interface, and being private is a lesser deal.

Regarding the logic, i would like to see it first, so that we can compare is actually simpler.
This was just porting over what was before somewhere else, i m not attached to it.

Just dump how you would write this whole method in a comment.

Reply:
```ts
// expose as const, can be reused by other parts
    const CORNER_ORIGIN = {
      tl: new Point(-0.5, -0.5),
      tr: new Point(0.5, -0.5),
      br: new Point(0.5, 0.5),
      bl: new Point(-0.5, 0.5),
    };

// expose in ObjectGeometry
  containsViewportPoint(point: Point, padding = 0) {
    const vpt = this.getViewportTransform();
    const paddingVector = new Point().scalarMultiply(padding * 2);
    const [tl, tr, br, bl] = this.getCoords();
    const viewportCoords = (
      Object.entries({ tl, tr, br, bl }) as [
        keyof typeof CORNER_ORIGIN,
        Point
      ][]
    ).map(([key, point]) =>
      sendPointToPlane(point, vpt).add(
        paddingVector.multiply(CORNER_ORIGIN[key])
      )
    );

    return Intersection.isPointInPolygon(point, viewportCoords);
  }

```

Reply:
There is a lot to say about this.

Is better explained in person but i can summarize that Object geometry is done with padding and viewport. We just remove the concept that padding is part of the object geometry, is just a selection concern, and saying 'containsViewportPoint' and adding back padding is rolling back the change.

Also in this example i don't understand who is rotating the padding vector. I see we are using a direction vector to correctly move the point in that direction, but where is the rotation?

Reply:
> but where is the rotation?

I missed that

> We just remove the concept that padding is part of the object geometry, is just a selection concern, and saying 'containsViewportPoint' and adding back padding is rolling back the change.

I have much to say about that. I have said that geometry must occur in the viewport but you did not take that into account so I don't see as rolling back

Reply:
I don't remember you saying that, or we wouldn't have remove the absolute from everywhere.
Anyway, next meeting then.
I l freeze everything meanwhile